doc/microsite/src/main/tut/distage/distage-testkit.md

```diff
@@ -1,14 +1,518 @@
+The `distage-testkit` simplifies pragmatic pure-FP testing. The `DistageSpecScalatest` are
+[ScalaTest](https://www.scalatest.org/) base classes for effect types of `Identity`, `F[_]`,
+`F[+_, +_]` and `F[-_, +_, +_]`. These provide an interface similar to ScalaTest's
+[`WordSpec`](http://doc.scalatest.org/3.1.0/org/scalatest/wordspec/AnyWordSpec.html). However,
+`distage-testkit` has additional capabilities: first class support for effect types as well as
+dependency injection.
 
-Further documentation still TBD: [Issue #820](https://github.com/7mind/izumi/issues/820)
+Usage of `distage-testkit` generally follows these steps:
+
+1. Select a base class corresponding to the effect type:
+    - No effect type - @scaladoc[`DistageSpecScalatest[Identity]`](izumi.distage.testkit.scalatest.DistageSpecScalatest)
+    - `F[_]` - @scaladoc[`DistageSpecScalatest[F]`](izumi.distage.testkit.scalatest.DistageSpecScalatest)
+    - `F[+_, +_]` - @scaladoc[`DistageBIOSpecScalatest[F]`](izumi.distage.testkit.scalatest.DistageBIOSpecScalatest)
+    - `F[-_, +_, +_]` - @scaladoc[`DistageBIOEnvSpecScalatest[F]`](izumi.distage.testkit.scalatest.DistageBIOEnvSpecScalatest)
+2. Override the `config` method to customize the @scaladoc[`TestConfig`](izumi.distage.testkit.TestConfig)
+3. Establish test case context using [`should`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/ShouldVerb.html),
+   [`must`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/MustVerb.html),
+   and [`can`](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/verbs/CanVerb.html).
+4. Introduce test cases using one of the `in` methods. Test cases can have a variety of forms. From
+   pure functions returning an
+   [assertion](https://www.scalatest.org/scaladoc/3.2.0/org/scalatest/Assertions.html) to effectful
+   functions with dependencies.
+    - No effect type / `Identity` - @scaladoc[`in`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$LowPriorityIdentityOverloads)
+    - @scaladoc[`in` for `F[_]`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper)
+    - @scaladoc[`in` for `F[+_, +_]`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper2)
+    - @scaladoc[`in` for `F[-_, +_, +_]`](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper3)
+    - Magnet for test cases dependent on injectables: @scaladoc[`ProviderMagnet`](izumi.distage.model.providers.ProviderMagnet)
+
+### API Overview
+
+To demonstrate the usage of `distage-testkit` we'll consider a hypothetical game scoring system. This
+will have a model, logic and service which we'll then define test cases to verify.
+
+The highest value tests to develop, [in our experience](https://www.youtube.com/watch?v=CzpvjkUukAs),
```

Comment:
Might better link to [blog post](https://blog.7mind.io/constructive-test-taxonomy.html) here or below when new terminology starts, and as the blog post is complimentary and blogs are more popular than videos.

Reply:
Ah! Thanks. Very useful. Any other references on testkit to include? I'll at least collect them in `references` if I can't find a way to integrate into the docs

Reply:
@coreyoconnor 
I think that's the only blog post, but in release notes there were more detailed descriptions of some of the new testkit features:
* memoization environment merging - https://github.com/7mind/izumi/releases/tag/v0.10.9
* `TestConfig#forcedRoots`, `#parallelEnvs`, zio environment tests - https://github.com/7mind/izumi/releases/tag/0.10.2

Reply:
Ah, slides for ["Hyper-pragmatic pure FP testing with distage-testkit"](https://www.slideshare.net/7mind/hyperpragmatic-pure-fp-testing-with-distagetestkit) & ["Scala, Functional Programming and Team Productivity"](https://www.slideshare.net/7mind/scala-functional-programming-and-team-productivity) both show examples of `distage-testkit` usage.